### PR TITLE
FrameTransformer implements tf2::BufferCoreInterface and tf2_ros::AsyncBufferInterface

### PR DESCRIPTION
**Pull request by jacobperron**
Requires https://github.com/ros2/geometry2/pull/121
Unblocks https://github.com/ros2/rviz/pull/375

The idea is to have `rviz_common::transformation::FrameTransformer` plugins implement the interfaces required by `tf2_ros::MessageFilter`. This will enable us to pass any frame transformers to the TF message filter.

---

I'm opening this for visibility. The implementation is subject to change pending the review of https://github.com/ros2/geometry2/pull/121. I also would like to refactor this so that we avoid the extra conversion to/from rviz and tf2 types in the case we are using a TFFrameTransformer.

### CODE REVIEW

**Comment by jacobperron** [comment]
Ready for review.

File: rviz_default_plugins/package.xml [hunk 1]
```diff
@@ -33,6 +33,9 @@
   <depend>resource_retriever</depend>
   <depend>rviz_common</depend>
   <depend>rviz_rendering</depend>
+  <depend>tf2</depend>
+  <depend>tf2_ros</depend>
+  <depend>tf2_geometry_msgs</depend>
```
Comment (wjwwood) [inline rviz_default_plugins/package.xml:37]: nitpick: Alphabetical ordering (maybe in a few other places too)

File: rviz_default_plugins/CMakeLists.txt [hunk 2]
```diff
@@ -66,6 +66,9 @@ find_package(nav_msgs REQUIRED)
 find_package(pluginlib REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(resource_retriever REQUIRED)
+find_package(tf2 REQUIRED)
+find_package(tf2_ros REQUIRED)
+find_package(tf2_geometry_msgs REQUIRED)
```
Comment (wjwwood) [inline rviz_default_plugins/CMakeLists.txt:70]: here

**Review by wjwwood — APPROVED** [review]

File: rviz_default_plugins/package.xml (hunk 1 above)
Comment (jacobperron) [reply rviz_default_plugins/package.xml:37]: Fixed df50f8c

File: rviz_default_plugins/CMakeLists.txt (hunk 2 above)
Comment (jacobperron) [reply rviz_default_plugins/CMakeLists.txt:70]: df50f8c

**Comment by jacobperron** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/7612/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/3732/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/6225/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/7433/)

**Review by wjwwood — APPROVED** [review]

**Pull request merged**